clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy

```diff
@@ -130,4 +127,48 @@ class SecurityGroupIngressConverter {
       }
     }.flatten().unique()
   }
+
+  static Map<String, List> computeLatestIpRules(List<IpPermission> newList, List<IpPermission> existingRules) {
+    List<IpPermission> tobeAdded = new ArrayList<>()
+    List<IpPermission> tobeRemoved = new ArrayList<>()
+    List<IpPermission> tobeUpdated = new ArrayList<>()
+    List<IpPermission> filteredList = newList.findAll { elements -> elements.userIdGroupPairs.size() == 0 }
+    List<IpPermission> filteredExisistingRuleList = existingRules.findAll { elements -> elements.userIdGroupPairs.size() == 0 }
+    filteredList.forEach({ newListEntry ->
+      IpPermission match = findIpPermission(filteredExisistingRuleList, newListEntry)
+      if (match) {
+        if (newListEntry.ipv4Ranges.collect { it.description }.find() != null
```

Comment:
i think this would read better if you had (for both of these obv but github won't let me do that)
```suggestion
        if (newListEntry.ipv4Ranges.collect { it.description }.any()
```

all you care is there is at least one description, right?